### DISCUSSION TITLE
Upgrading ubuntu 20 to ubuntu 22, result in bad gateway

### DISCUSSION THREAD

**Discussion by lexisman** [Ylianst/MeshCentral#5588] [Q&A]
I mistakenly upgraded my ubuntu20 to ubuntu22 and now I get error 502 bad gateway. Anyone encounter this issue and got a fixed? Thanks.

**Comment by dinger1986** [comment]
were you using mongodb?

**Comment by lexisman** [comment]
Yes.

**Reply by dinger1986** [reply]
need to install it again, I am sure theres a dependency missing https://www.cherryservers.com/blog/install-mongodb-ubuntu-22-04

**Comment by lexisman** [comment]
you mean install just the mongodb?

**Reply by dinger1986** [reply]
yeah if you do `systemctl status mongod` it will probably say unrecognised

**Comment by lexisman** [comment]
This is terrible mistake on my part to upgrade ubuntu to 22.04, I am still trying to find a way, following all the suggestions on mongodb community, but nothing works. One question, Could I instead change the config.json to use the NEDB? Will there be big implications?

**Reply by dinger1986** [reply]
Is it not installing?

If you have backups you could restore.

I am sure I got mongo running on 22.04 I'll try and spin one up quickly to check

**Reply by lexisman** [reply]
Unfortunately I have no backup of the ubuntu system itself. I only have a backup of the meshcentral files. I would appreciate it if you could post what you did to install and run it. Thanks.

PS. As most people who were trying to install mongodb on ubuntu22.04 this is the error: core-dump: STATUS 4/ILL

**Reply by dinger1986** [reply]
https://www.mongodb.com/community/forums/t/how-to-install-mongodb-6-0-on-ubuntu-22-04/176976/6

Has the steps to do it, its missing libssl or something like that

**Reply by lexisman** [reply]
I am stuck with this error on whatever version of mongodb I install. "
![Screenshot 2023-11-28 105334](https://github.com/Ylianst/MeshCentral/assets/38063252/c13fe929-35a9-4e5e-b450-d2276b11e6e8)
![Screenshot 2023-11-28 105455](https://github.com/Ylianst/MeshCentral/assets/38063252/d61f0725-ec1d-4ce8-91b8-bb18ecbc822d)
![Screenshot 2023-11-28 105605](https://github.com/Ylianst/MeshCentral/assets/38063252/5f6c6555-fd1e-49b5-8f6a-1618330e1e53)
I tried googling and I could not find a definite answer. I also noticed that this is a very common problem from other users. Maybe others here have encountered this issue.

**Reply by si458** [reply]
if you do `dpkg -l | grep mongo` what version gets outputted?

**Reply by lexisman** [reply]
![Screenshot 2023-11-28 110410](https://github.com/Ylianst/MeshCentral/assets/38063252/d5446c62-da50-46ac-8c7e-5c82b3789612)

**Reply by si458** [reply]
you could follow the offical docs to reinstall it but you might need to uninstall the old package first if ur comfy with that?
(you might lose data im not 100% sure as never used mongodb)
https://www.mongodb.com/docs/v6.0/tutorial/install-mongodb-on-ubuntu/

**Reply by si458** [reply]
oh dear you might be stuffed! unless u can install an old version of mongodb
MongoDB 5.0+ requires use of the AVX instruction set, available on [select Intel and AMD processors.](https://en.wikipedia.org/wiki/Advanced_Vector_Extensions#CPUs_with_AVX)

**Reply by si458** [reply]
you could try installing version 4.4+ as that supports 22.04 with avx
https://www.mongodb.com/docs/v4.4/tutorial/install-mongodb-on-ubuntu/
package repo shows deb files for jammy (22.04) - https://repo.mongodb.org/apt/ubuntu/dists/jammy/mongodb-org/4.4/multiverse/binary-amd64/

**Comment by si458** [comment]
I would first check u have a current version (18 at least) of node installed `node -v`
Then check ur mongodb is running
Then try stopping meshcentral and start it and check the mesherrors.txt for errors

**Reply by lexisman** [reply]
Can you tell me how to update the node to latest version? I only have version 16. An I remember it has to be  installed manually if I am correct. Sorry I found to update it. So I have version 20 now of node.

**Reply by si458** [reply]
So if u have 20 now, it should start OK, if u do `systemctl status meshcentral` what's the output?

**Comment by lexisman** [comment]
Thanks for the help.
